test/e2e/smoke.test.js

```diff
@@ -0,0 +1,111 @@
+/* global it afterEach expect */
+import puppeteer from 'puppeteer'
+import ms from 'milliseconds'
+
+// Which deployment to run the tests against
+const appUrl = process.env.URL || 'http://localhost:3000'
+const debug = process.env.DEBUG === 'true'
+
+console.log('Testing', appUrl)
+
+// puppeteer.launch opts
+// see: https://github.com/GoogleChrome/puppeteer/blob/v1.3.0/docs/api.md#puppeteerlaunchoptions
+const headless = !debug
+const slowMo = debug ? 100 : undefined
+const noSandbox = process.env.NO_SANDBOX === 'true'
+const args = noSandbox ? ['--no-sandbox'] : undefined
+
+// track browser instances so we can close them all after the test.
+let browsers = []
+
+async function createBrowser () {
+  const browser = await puppeteer.launch({headless, slowMo, args})
+  browsers.push(browser)
+  return browser
+}
+
+afterEach(() => {
+  browsers.forEach(b => b.close())
+})
+
+it('Create a pad', async (done) => {
```

Comment:
Good spot. I was getting `unhandedRejectedPromise` warnings rather than test failures, which is why these tests are a weird mixture of `async await` and `done` callbacks. I need to dig into that.

Reply:
I've tried to recreate the madness and didn't get the same issue so have fixed up the tests to be more sane as per your suggests. much nice.